src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -260,7 +267,7 @@ class InstanceOpFactoryImpl(
           } else {
             Seq(Task.Id.forInstanceId(reservedInstance.instanceId))
           }
-          originalIds.map(ti => Task.Id.forResidentTask(ti)).to[Seq]
+          originalIds.map(ti => Task.Id.withIncarnationCount(ti)).to[Seq]
```

Comment:
Yeah, It should not be up to this logic whether an incarnation count is there or not.

Reply:
```scala
originalIds.map(ti => ti.increaseIncarnation()).to[Seq]
```
see https://github.com/mesosphere/marathon/pull/6660/

Reply:
I think what's being said here is that incarnation count should always be there.

Reply:
(the method `forResidentTask` was renamed to `withIncarnationCount`)

Reply:
It is some nit picking. IMHO the logic should be part of the id not some methods outside of it. It's not a big deal.